src/SSHDebugPS/CommandRunner.cs

```diff
@@ -177,13 +201,16 @@ protected virtual void ReadLoop(StreamReader reader, CancellationToken token, bo
                         return; // end of stream
                     }
 
-                    action(task.Result);
+                    action(line);
                 }
             }
             catch (Exception e)
             {
-                ReportException(e);
-                Dispose();
+                if (this.IsDisposeStarted) // ignore exceptions if we are disposing
+                {
+                    ReportException(e);
```

Comment:
Is the `ReportException` supported to be in the `else` clause? I'm confused since the comment says `// ignore exceptions if we are disposing`.

Reply:
I had my `if` condition backwards.